packages/gcf-utils/src/logging-octokit-plugin.ts

```diff
@@ -0,0 +1,245 @@
+    case GitHubActionType.ISSUE_CREATE_LABEL:
+    case GitHubActionType.ISSUE_DELETE_LABEL:
+      return String(options.name);
+    case GitHubActionType.ISSUE_UPDATE_LABEL:
+      return `${options.current_name} to ${options.name}`;
+    case GitHubActionType.ISSUE_UPDATE:
+      return 'updated: ' + hasIssueUpdateProps.join(',');
+    case GitHubActionType.ISSUE_CREATE:
+      return String(options.title);
+    case GitHubActionType.PR_DISMISS_REVIIEW:
+      return `dismiss ${options.review_id}: ${options.message}`;
+    case GitHubActionType.PR_MERGE:
+    case GitHubActionType.PR_UPDATE_BRANCH:
+    default:
+      return 'NONE';
+  }
+}
+
+/**
+ * Parses the outgoing GitHub request to determine the details of the action being taken
+ * @param options options from outgoing request
+ */
+function parseActionDetails(options: {
+  [key: string]: string | number;
+}): GitHubActionDetails {
+  const actionType: GitHubActionType =
+    ActionEndpoints[options.url][options.method] || GitHubActionType.UNKNOWN;
+
+  const details: GitHubActionDetails = {};
+  details.value = parseActionValue(actionType, options);
+
+  if (options['issue_number']) {
+    details.destObjType = GitHubObjectType.ISSUE;
+    details.destObjId = options['issue_number'];
+  } else if (options['pull_number']) {
+    details.destObjType = GitHubObjectType.PR;
+    details.destObjId = options['pull_number'];
+  }
+
+  details.repoName = String(options['repo']);
+  details.repoOwner = String(options['owner']);
+  details.type = actionType;
+
+  return details;
+}
+
+/**
+ * Log a GitHub action as a GitHubActionLog
+ * @param logger logger instance
+ * @param details action details
+ */
+function logGithubAction(logger: MetricLogger, details: GitHubActionDetails) {
+  if (!details.type || details.type === GitHubActionType.UNKNOWN) {
+    // don't log unknown action types
+    return;
+  }
+  const general = {
+    type: details.type,
+    value: details.value || 'UNKNOWN',
+    destination_repo: {
+      repo_name: details.repoName || 'UNKNOWN',
+      owner: details.repoOwner || 'UNKNOWN',
+    },
+  };
+
+  const destination_object =
+    details.destObjType || details.destObjId
+      ? {
+          destination_object: {
+            object_type: details.destObjType || GitHubObjectType.UNKNOWN,
+            object_id: details.destObjId || 'UNKNOWN',
+          },
+        }
+      : {};
+
+  const githubAction: GitHubActionLog = {
+    action: {...general, ...destination_object},
+  };
+  logger.metric(githubAction);
+}
+
+/**
+ * Hooks into outgoing requests from Octokit to log metrics
+ */
+module.exports = (
+  octokit: Octokit,
+  pluginOptions: {customLogger?: MetricLogger}
+) => {
+  const octoLogger = pluginOptions.customLogger || logger;
+
+  octokit.hook.before('request', async options => {
+    const actionDetails: GitHubActionDetails = parseActionDetails(options);
+
+    logGithubAction(octoLogger, actionDetails);
+  });
+
+  (octokit as Octokit & {
```

Comment:
Are you changing the type and properties of an octokit instance? Will this have a side-effect on users that assume their octokit's properties aren't modified?

Reply:
The cast is only local to that statement so I'm not changing the type returned to the bots. While this does add new properties to the octokit instance (as does the rest of the plugin) it maintains backward compatibility